File: DSA/DevvratKhandal_590014906/Day_22/Question_1.c
#include <stdio.h>

void moveZeroes(int nums[], int n) {
    int index = 0;

    // Move non-zero elements
    for (int i = 0; i < n; i++) {
        if (nums[i] != 0) {
            nums[index] = nums[i];
            index++;
        }
    }

    // Fill remaining positions with 0
    while (index < n) {
        nums[index] = 0;
        index++;
    }
}

int main() {
    int n;

    printf("Enter number of elements: ");
    scanf("%d", &n);

    int nums[n];

    printf("Enter array elements: ");
    for (int i = 0; i < n; i++) {
        scanf("%d", &nums[i]);
    }

    moveZeroes(nums, n);

    printf("Array after moving zeroes: ");
    for (int i = 0; i < n; i++) {
        printf("%d ", nums[i]);
    }

    return 0;
}

File: DSA/ShlokMadan_590017851/Day_22/Question1.java
public class Question1 {
    public static void moveZeroes(int[] nums)
    {
        int index=0;
        for(int i=0;i<nums.length;i++)
            {
            if(nums[i]!=0)
                {
                int temp=nums[index];
                nums[index]=nums[i];
                nums[i]=temp;
                index++;
            }
        }
    }
    public static void main(String[] args)
    {
        int[] nums={0,1,0,3,12};
        moveZeroes(nums);
        System.out.print("[");
        for(int i=0;i<nums.length;i++)
            {
            System.out.print(nums[i]);
            if(i!=nums.length-1)
                System.out.print(",");
        }
        System.out.println("]");
    }
}